app/process_kb_results.py

```diff
@@ -108,24 +112,38 @@ def facet_query_string(query, terms, facets, type_map):
 
     # Add search query if it exists
     qt = ""
-    if query is not "":
+    if query:
         qt = f'({query})'
 
-    if query is not "" and len(t) > 0:
+    if query and t:
         qt += " AND "
 
     # Add the brackets and OR and AND parameters
     for k in t:
-        qt += type_map[k][0] + ":("  # facet term path and opening bracket
-        for l in t[k]:
-            qt += f"({l})"  # bracket around terms incase there are spaces
-            if l is not t[k][-1]:
-                qt += " OR "  # 'OR' if more terms in this facet are coming
-            else:
-                qt += ") "
+        if k == "datasets":
+            needParentheses = (qt or len(t) > 1) and (len(t[k]) > 1)
+            if needParentheses:
+                qt += "("
+            for l in t[k]:
+                if l == "scaffolds":
+                    qt += "scaffolds.object.mimetype.name:((inode%2fvnd.abi.scaffold+directory))"
```

Comment:
The following query_string will give more accurate result after the SciCrunch update -
"objects.additional_mimetype.name:((inode%2fvnd.abi.scaffold) AND (file))"

Reply:
Done in commit 3aaaea8.